### PR TITLE
implemented @groupclaes/fastify-elastic mssql plugin

### PR DESCRIPTION
**Pull request by JamieVangeysel**

### CODE REVIEW

**Review by JamieVangeysel — COMMENTED** [review]
lgtm!

File: src/controllers/document.controller.ts [hunk 1]
```diff
@@ -38,7 +44,8 @@ export default async function (fastify: FastifyInstance) {
       return reply.fail({ role: 'missing permission' }, 403)
 
     try {
-      const repo = new Document(request.log)
+      const pool = await fastify.getSqlPool(env['DB_NAME'] ?? 'PCM')
```
Comment (OhSoooLucky) [inline src/controllers/document.controller.ts]: Maybe default to DB_NAME in getSqlPool, and allow you to configure a default DB in the Fastify wrapper config.

This should reduce duplicate code like this

File: src/index.ts [hunk 2]
```diff
@@ -46,15 +42,4 @@ async function main(config: any): Promise<FastifyInstance | undefined> {
   await fastify.listen({ port: +(env['PORT'] ?? 80), host: '::' })
 
   return fastify
-}
-
-['SIGTERM', 'SIGINT'].forEach(signal => {
```
Comment (OhSoooLucky) [inline src/index.ts:51]: Why remove this?

File: src/repositories/access-log.repository.ts [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
+  constructor(logger: FastifyBaseLogger, pool: sql.ConnectionPool) {
+    this._logger = logger
+    this._pool = pool
+  }
 
   async get(user_id?: string) {
-    const r = new sql.Request(await db.get(DB_NAME))
+    const r = new sql.Request(this._pool)
     r.input('user_id', sql.Int, user_id)
-    this._logger.debug({ sqlParam: { user_id }, sqlDb: DB_NAME, sqlSchema: this.schema, sqlProc: '[uspGetAccessLogs]' }, 'running procedure')
```
Comment (OhSoooLucky) [inline src/repositories/access-log.repository.ts:16]: why remove db name? Cuz because of only 1 db being used?

**Review by OhSoooLucky — APPROVED** [review]

File: src/index.ts (hunk 2 above)
Comment (JamieVangeysel) [reply src/index.ts:51]: this is moved into index.ts file in order to enable unit testing

File: src/repositories/access-log.repository.ts (hunk 3 above)
Comment (JamieVangeysel) [reply src/repositories/access-log.repository.ts:16]: This variable is no longer accessible in the repository

**Pull request merged**